src/bindings/js/node/include/async_reader.hpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <napi.h>
+
+#include "model_wrap.hpp"
+#include "openvino/runtime/core.hpp"
+#include "read_model_args.hpp"
+
+class ReaderWorker : public Napi::AsyncWorker {
+public:
+    /**
+     * @brief Constructs ReaderWorker class that is responisible for reading the model asynchronously.
+     * @param info contains passed arguments. Can be empty.
+     */
+    ReaderWorker(const Napi::Env& env, ReadModelArgs* args)
+        : Napi::AsyncWorker{env, "ReaderWorker"},
+          _deferred{env},
+          _model{} {
```

Comment:
```suggestion
          _model{}, _args{args} {
          OPENVINO_ASSERT(_args, "Invalid pointer to ReadModelArgs.");
```

Detail: member can be initialized first if it has got wrong value then throw.